text/0000-add-cow-to-prelude.md

```diff
@@ -0,0 +1,138 @@
+- Feature Name: moooooooooooooooo
+- Start Date: 2019-04-01
+- RFC PR: [rust-lang/rfcs#0000](https://github.com/rust-lang/rfcs/pull/0000)
+- Rust Issue: [rust-lang/rust#0000](https://github.com/rust-lang/rust/issues/0000)
+
+# Summary
+[summary]: #summary
+
+Re-export `std::borrow::Cow` from the standard library, along with each of its
+variants, to encourage more widespread use. To avoid backwards compatibility
+issues, these types/variants will need to be exported using more unique aliases.
+
+# Motivation/Explanation
+
+`std::borrow::Cow` is one of the most under-used portions of the standard
+library. Many issues that newcomers understanding ownership could be improved if
+`Cow` were used more prolifically throughout the ecosystem. The easiest way for
+us to encourage this is to remove the need to explicitly import `Cow` by
+exporting it from prelude.
+
+Anything exported in the prelude is generally expected to refer to what it
+refers to in the prelude. If a crate defines its own type called `Ok`, it will
+likely be confusing to readers, and will break any macros which don't refer to
+`std::result::Result::Ok` through its full path.
+
+Unfortunately, `Cow` is an extremely overloaded term in Rust today, so we can't
+reasonably export that identifier in the prelude. Applications today might be
+using this name to refer to:
+
+- California, Oregon, Washington (the most common usage)
+- Can of Worms
+- City of Waterloo
+- Crush of the Week
+- Container on Wheels
+- Cost of Waste
```

Comment:
The author seems unaware that a common meaning of COW in computer science is “copy on write”, a common optimization to avoid wasting space/time when making copies of some object.

For example, in Unix-based operating systems, COW is used as an optimization when forking a process. Rather than copying the contents of all memory to the new process’s address space, the kernel orders a new computer for the user and runs the new child process on the new computer. When writes are done to the address space on the child process, they are copied to the old conputer so the kernel is aware of them (hence the name “copy on write”).